src/UXClient/Components/Marker/Marker.ts

```diff
@@ -0,0 +1,274 @@
+
+
+    private setTimeLabel (closestTime: number) {
+        let values: Array<any> = this.chartComponentData.timeMap[closestTime];
+        if (values == undefined || values.length == 0) {
+            return;
+        }
+        let firstValue = values[0].dateTime;
+        let secondValue = new Date(values[0].dateTime.valueOf() + (values[0].bucketSize != null ? values[0].bucketSize : 0));
+        let timeFormat = Utils.timeFormat(this.chartComponentData.usesSeconds, this.chartComponentData.usesMillis, 
+            this.chartOptions.offset, this.chartOptions.is24HourTime, null, null, this.chartOptions.dateLocale);
+        let dateToTime = (t) => ((timeFormat(t).split(" ") && timeFormat(t).split(" ").length > 1) ? timeFormat(t).split(" ")[1] : '');
+        let text = dateToTime(firstValue) + " - " + dateToTime(secondValue);
+        let self = this;
+
+        this.timeLabel.text(text);
+
+        this.closeButton = this.timeLabel.append("button")
+            .attr("aria-label", this.getString("Delete marker at") + ' ' + text) 
+            .classed("tsi-closeButton", true)
+            .on("click", function () {
+                d3.select(d3.select(this).node().parentNode.parentNode).remove();
+                self.onChange(true, false);
+            });
+
+        let markerLeft: number = Number(this.markerContainer.style("left").replace("px", ""));
+        let timeLabelWidth: number = Math.round(this.timeLabel.node().getBoundingClientRect().width);
+        let minLeftPosition = this.marginLeft + LINECHARTXOFFSET;
+        let width = this.x.range()[1] - this.x.range()[0];
+        let maxRightPosition = width + this.marginLeft + LINECHARTXOFFSET + this.ADDITIONALRIGHTSIDEOVERHANG;
+        let calculatedLeftPosition = markerLeft - (timeLabelWidth / 2);
+        let calculatedRightPosition = markerLeft + (timeLabelWidth / 2);
+        let translate = "translateX(calc(-50% + 1px))";
+        if (calculatedLeftPosition < minLeftPosition) {
+            translate = "translateX(-" + Math.max(0, markerLeft - minLeftPosition) + "px)";
+        }
+        if (calculatedRightPosition > maxRightPosition) {
+            translate = "translateX(calc(-50% + " + (maxRightPosition - calculatedRightPosition) + "px))";
+        }
+
+        this.timeLabel
+            .style("-webkit-tranform", translate)
+            .style("transform", translate);
+    }
+
+    public focusCloseButton () {
+        this.closeButton.node().focus();
+    }
+
+    private isMarkerInRange (millis: number) {
+        let domain = this.x.domain();
+        return !(millis < domain[0].valueOf() || millis > domain[1].valueOf());
+    }
+
+    public destroyMarker () {
+        if (this.markerContainer) {
+            this.markerContainer.remove();
+        }
+        this.markerContainer = null;
+    }
+
+    public render (timestampMillis: number, chartOptions: ChartOptions, chartMargins: any, chartComponentData: any, x: any, chartHeight: number, marginLeft: number, 
```

Comment:
i think perhaps render(timestampMillis: number, chartOptions: ChartOptions, chartComponentData: any, {chartMargins: any, x: any}) could make sense here.  so 4 parameters, where the fourth one is a destructured object catch all for the non-strongly typed params